src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +325,37 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module, IjProjectConfig projectConfig, ProjectFilesystem projectFilesystem) {
+    Path projectRootPath = Paths.get(projectConfig.getProjectRoot());
+    Path extraModuleBasePath =
+        projectRootPath
+            .resolve("buck-out")
+            .resolve("extra-intellij-output")
```

Comment:
I suppose you can move this location to a config parameter and change logic to enable with extra dependencies if path is not empty.